src/main/kotlin/org/dif/crypto/JWS.kt

```diff
@@ -44,9 +45,16 @@ fun sign(payload: String, key: Key): String {
         .serialize()
 }
 
-fun verify(jws: JWSObjectJSON, key: Key): Message {
+fun verify(jws: JWSObjectJSON, key: Key): VerifyResult {
     val jwk = key.jwk
 
+    val signAlg: SignAlg = when (val alg = jws.header.algorithm) {
+        JWSAlgorithm.ES256 -> SignAlg.ES256
+        JWSAlgorithm.ES256K -> SignAlg.ES256K
+        JWSAlgorithm.EdDSA -> SignAlg.ED25519
```

Comment:
What if for example `jws.header.algorithm` == JWSAlgorithm.EdDSA, and `getJWSAlgorithm(jwk)` == JWSAlgorithm.ES256K?